tests/shared/utils/test_common.py

```diff
@@ -140,3 +141,44 @@ async def my_function():
             await asyncio.sleep(0)
 
         await my_function()
+
+
+@pytest.mark.parametrize(
+    "module_path, lookup_path, outcome",
+    [
+        ("rasa.shared.core.domain.Domain", None, "Domain"),
+        # lookup in locals()
+        ("Domain", None, "Domain"),
```

Comment:
this test fails. I am not sure how the function is supposed to work with `globals()` (cf. [docs](https://docs.python.org/3/library/functions.html#globals)). I think we assumed it did something, and actually it doesn't work.

Reply:
or do we expect that the `globals()` are updated after the function is called successfully, making the subsequent calls work? 🤷🏻

Reply:
when you are saying "this test fails" do you mean for the parameters in the line we are talking about here or do all of these fail?

Reply:
I'm talking about these specific parameters. All the other tests pass

Reply:
cc @wochinge I suspect that there is a use case that never works in this function (or I am missing something): I wrote a test case with that should go down [this branch](https://github.com/RasaHQ/rasa/blob/230c773d4b0ad0c1dc158f8efa1d8252fc15eb42/rasa/shared/utils/common.py#L43) and return a loaded class, but it never works. (cf. my two first comments in this thread).

Reply:
thanks for adding the new tests 💯 
Using `locals()` as fallback doesn't make any sense in my opinion. Also `globals` can't work as it will only have the globals [of the module](https://docs.python.org/3/library/functions.html#globals) which the `class_from_module_path`  is defined in. As this is a pure utils class there is nothing of interest in the `globals`. In my opinion two things make sense:
1. Remove this part (preferred option) 
2. Make it possible to pass the callers `globals` (less preferred as no code is currently using this)

What do you think?

Reply:
I think removing it is best 👍🏻 if we need the 2. at some point, we can always implement it later (which I doubt...)